tests/native-lib/fail/struct_not_extern_c.rs

```diff
@@ -0,0 +1,19 @@
+// Only works on Unix targets
+//@ignore-target: windows wasm
+//@only-on-host
+
+#![allow(improper_ctypes)]
+
+pub struct PassMe {
+    pub value: i32,
+    pub other_value: i16,
+}
+
+extern "C" {
+    fn pass_struct(s: PassMe) -> i32;
+}
+
+fn main() {
+    let pass_me = PassMe { value: 42, other_value: 1337 };
+    unsafe { pass_struct(pass_me) }; //~ ERROR: Undefined Behavior: passing a non-#[repr(C)] struct over FFI
```

Comment:
Is this correct? I think it's UB, but maybe if the user intentionally sets `#[allow(improper_ctypes)]` we might just want to report this as unsupported instead of UB?